src/Presence.js

```diff
@@ -73,6 +73,11 @@ class Presence {
         }
         var old_state = this.state;
         this.state = newState;
+
+        if (MatrixClientPeg.get().isGuest()) {
+            return; // don't try to set presence when a guest; it won't work.
+        }
+
```

Comment:
Synapse doesn't let us currently so it *spams* errors into the console which makes me sad. I can enable it though if you prefer.